### PR TITLE
Added request object to server response.

### PR DESCRIPTION
**Pull request by timmckenzie**
Will help with dealing with multiple requests and responses with more generic handlers.

### CODE REVIEW

File: lib/http.js [hunk 1]
```diff
@@ -973,6 +973,7 @@ OutgoingRequest.prototype = Object.create(OutgoingMessage.prototype, { construct
 
 OutgoingRequest.prototype._start = function _start(stream, options) {
   this.stream = stream;
+  this.options = options;
```
Comment (nwgh) [inline lib/http.js]: This seems totally unrelated to the commit message.

File: lib/http.js [hunk 2]
```diff
@@ -998,8 +999,8 @@ OutgoingRequest.prototype._start = function _start(stream, options) {
   this.headersSent = true;
 
   this.emit('socket', this.stream);
-
   var response = new IncomingResponse(this.stream);
+  response.req = this;
```
Comment (nwgh) [inline lib/http.js]: Call this "response.request" - needlessly short names like "req" should be avoided.

File: lib/http.js (hunk 1 above)
Comment (timmckenzie) [inline lib/http.js]: I suppose that this has more to do with using more generic event handlers.  The options for request can contain additional options which would not be copied to the request object.

File: lib/http.js (hunk 2 above)
Comment (timmckenzie) [inline lib/http.js]: I picked response.req because that is the same as the node HTTP library.  https://github.com/joyent/node/blob/master/lib/_http_client.js#L423 

I am not a huge fan on needlessly short name either, but I would promote compatibility with the core HTTP.

**Comment by nwgh** [comment]
Alright, I'm sold. Thanks!
